Applications/Spire/Source/Ui/ListView.cpp

```diff
@@ -207,11 +207,19 @@ void ListView::append_query(const QString& query) {
   if(!m_items.empty()) {
     auto start = m_current_model->get_current().get_value_or(-1);
     auto i = (start + 1) % static_cast<int>(m_items.size());
+    auto is_repeated_query = m_query.count(m_query.at(0)) == m_query.count();
+    auto short_match = boost::optional<int>();
     while(i != start) {
-      if(m_items[i]->m_item->isEnabled() && displayTextAny(
-          m_list_model->at(i)).toLower().startsWith(m_query.toLower())) {
-        set_current(i);
-        break;
+      if(m_items[i]->m_item->isEnabled()) {
+        if(displayTextAny(
+            m_list_model->at(i)).toLower().startsWith(m_query.toLower())) {
+          short_match = boost::none;
+          set_current(i);
+          break;
+        } else if(is_repeated_query && !short_match && displayTextAny(
+            m_list_model->at(i)).toLower().startsWith(m_query[0])) {
```

Comment:
This is computing ```displayTextAny(...)``` twice instead of reusing the calculation that must have been performed on line 214. Reuse as much of this as possible since these are not trivial operations.